modules/payment/src/Controller/CheckoutPaymentController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Drupal\commerce_payment\Controller;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OffsitePaymentGatewayInterface;
+use Drupal\Core\Access\AccessException;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class CheckoutPaymentController implements ContainerInjectionInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static();
+  }
+
+  /**
+   * Controller callback for offsite payments cancelled.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $commerce_order
+   *   The order.
+   */
+  public function checkoutPaymentCancel(OrderInterface $commerce_order) {
```

Comment:
what about cancelPaymentPage()? We tend to suffix these methods with Page, and it sounds more descriptive.